sdks/python/apache_beam/utils/multi_process_shared.py

```diff
@@ -0,0 +1,153 @@
+#
+# Licensed to the Apache Software Foundation (ASF) under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+"""Implements a shared object that spans processes.
+
+This object will be instanciated once per VM and methods will be invoked
+on it via rpc.
+"""
+# pytype: skip-file
+
+import logging
+import os
+import multiprocessing.managers
+import pickle
+import time
+import threading
+import uuid
+
+from typing import Any
+from typing import Callable
+from typing import Generic
+from typing import Optional
+from typing import TypeVar
+
+import fasteners
```

Comment:
Does this require a new dependency?

Reply:
Yes. It's a small one (and already in google). Updated the relevant places; thanks for the catch.

Reply:
Note also that we already transitively depend on in.